### PR TITLE
- New Adaptor - Inneractive

### PR DESCRIPTION
**Pull request by uriw-ia**
## Type of change

- [x] New bidder adapter

## Description of change

 New Inneractive PrebidJS Adaptor

- test parameters for validating bids
```
{
  bidder: 'inneractive',
  params: {
    appId: 'Inneractive_Superman_iPhone',
    spotType: 'rectangle',
    customParams: {
      portal: 7002
    }
  }
}
```
- contact email of the adapter’s maintainer
mobileweb@inner-active.com

- [x] official adapter submission

### CODE REVIEW

**Comment by aneuway2** [comment]
Hey @uriw-ia  are `appId` and `spotType` both required parameters for your adaptor? Might be good to submit a PR to have this documented on prebid.org :-)

https://github.com/prebid/prebid.github.io/tree/master/dev-docs/bidders

**Comment by uriw-ia** [comment]
Hi @aneuway2,

I've reopened a pull request on https://github.com/prebid/prebid.github.io/ with the documentation 
https://github.com/prebid/prebid.github.io/pull/200

(please also see: https://github.com/prebid/prebid.github.io/pull/195).

Thanks,
    Uri

**Comment by matthewlane** [comment]
Thanks for the adapter PR. I'm not able to get any bids back with the test parameters, are those still valid?

Also this adapter seems to be doing some analytic reporting, can you describe what's happening there?

**Comment by uriw-ia** [comment]
@matthewlane ,
Could you please try again? 
There seems to have been a temporary issue receiving the ad. 

Regarding analytics - our adapter is sending out the following events:

- "HBPreBidImpression" - for ad impression
- "HBPreBidClick" - for ad click
- "HBPreBidError" - for adapter JS error (unused)
- "HBPreBidNoAd" - for cases in which no ad was received from our servers
- "HBNoWin" - for cases in which an ad was received from our servers but didn't eventually win the bid

File: src/adapters/inneractive.js [hunk 1]
```diff
@@ -0,0 +1,488 @@
...
+  _checkIfBidWon(){
+    if (typeof $$PREBID_GLOBAL$$ === 'undefined') {
+      return;
+    }
+    const wonAdIds = [];
+    const numOfUnits = $$PREBID_GLOBAL$$.adUnits.length;
+    const _reportNoBidWon = () => this._getStoredBids().forEach(bid => !wonAdIds.includes(bid.adId) && Reporter.reportEvent('HBNoWin', bid.params));
+    let timer = null;
+    $$PREBID_GLOBAL$$.onEvent(EVENTS.AUCTION_END, () => timer = setTimeout(_reportNoBidWon, $$PREBID_GLOBAL$$.bidderTimeout), null);
+    $$PREBID_GLOBAL$$.onEvent(EVENTS.BID_WON, (bid) => {
```
Comment (matthewlane) [inline src/adapters/inneractive.js]: Please drop this event listener

**Comment by uriw-ia** [comment]
@matthewlane, sure thing.
The method listening for the event and the call to it, have been removed

**Comment by uriw-ia** [comment]
@matthewlane, the code is updated based on your comment. Thanks.

File: src/adapters/inneractive.js [hunk 2]
```diff
@@ -0,0 +1,462 @@
...
+    return headersData;
+  }
+};
+
+
+/**
+ * InnerActiveAdapter for requesting bids
+ * @class
+ */
+class InnerActiveAdapter{
```
Comment (mkendall07) [inline src/adapters/inneractive.js]: We havne't been pushing to use `class` but it seems to work ok and transpile.

**Pull request merged**